lib/egon/overcloud/undercloud_handle/node.rb

```diff
@@ -10,7 +10,18 @@ def list_nodes
     
     def create_node(node_parameters, create_flavor = false)
       node = service('Baremetal').nodes.create(node_parameters)
+      create_port({:node_uuid => node.uuid, :address => node_parameters[:address]})
+
       create_flavor_from_node(node) if create_flavor
+
+      node.set_provision_state('manage')
+      introspect_node(node)     
+      Thread.new {
```

Comment:
Nope; introspect_node works asynchronously, so this code continues as soon as the introspection process kicks off.  Then the Thread continues to check to see if introspection is complete, and once it is, changes the provision_state.